File: trains-gdx-core/src/main/java/com/github/caay2000/trains/Route.java
package com.github.caay2000.trains;

public class Route {

    private Position start;
    private Position end;

    public Route(Position start, Position end) {
        this.start = start;
        this.end = end;
    }

    public Position getStart() {
        return start;
    }

    public Position getEnd() {
        return end;
    }
}

File: trains-gdx-core/src/main/kotlin/com/github/caay2000/trains/world/entity/Route.kt
package com.github.caay2000.trains.world.entity

import com.github.caay2000.trains.world.location.City

class Route(var start: City, var end: City) {

    fun endRoute() {
        this.start = this.end.also { this.end = this.start }
    }
}